egs/swbd/s5c/local/chain/tuning/run_cnn_tdnn_1a.sh

```diff
@@ -0,0 +1,280 @@
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 11 ]; then
+  # Build a tree using our new topology. This is the critically different
+  # step compared with other recipes.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --context-opts "--context-width=2 --central-position=1" \
+      --cmd "$train_cmd" 7000 data/$train_set $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: creating neural net configs using the xconfig parser";
+
+  num_targets=$(tree-info $treedir/tree |grep num-pdfs|awk '{print $2}')
+  learning_rate_factor=$(echo "print 0.5/$xent_regularize" | python)
+
+  cnn_opts="l2-regularize=0.01"
+  ivector_affine_opts="l2-regularize=0.01"
+  tdnnf_first_opts="l2-regularize=0.01 dropout-proportion=0.0 bypass-scale=0.0"
+  tdnnf_opts="l2-regularize=0.01 dropout-proportion=0.0 bypass-scale=0.66"
+  linear_opts="l2-regularize=0.01 orthonormal-constraint=-1.0"
+  prefinal_opts="l2-regularize=0.01"
+  output_opts="l2-regularize=0.002"
+
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  input dim=100 name=ivector
+  input dim=40 name=input
+  # this takes the MFCCs and generates filterbank coefficients.  The MFCCs
+  # are more compressible so we prefer to dump the MFCCs to disk rather
+  # than filterbanks.
+  idct-layer name=idct input=input dim=40 cepstral-lifter=22 affine-transform-file=$dir/configs/idct.mat
+  linear-component name=ivector-linear $ivector_affine_opts dim=200 input=ReplaceIndex(ivector, t, 0)
+  batchnorm-component name=ivector-batchnorm target-rms=0.025
+  batchnorm-component name=idct-batchnorm input=idct
+  combine-feature-maps-layer name=combine_inputs input=Append(idct-batchnorm, ivector-batchnorm) num-filters1=1 num-filters2=5 height=40
+  conv-relu-batchnorm-layer name=cnn1 $cnn_opts height-in=40 height-out=40 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=64 
+  conv-relu-batchnorm-layer name=cnn2 $cnn_opts height-in=40 height-out=40 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=64
+  conv-relu-batchnorm-layer name=cnn3 $cnn_opts height-in=40 height-out=20 height-subsample-out=2 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=128
+  conv-relu-batchnorm-layer name=cnn4 $cnn_opts height-in=20 height-out=20 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=128
+  conv-relu-batchnorm-layer name=cnn5 $cnn_opts height-in=20 height-out=20 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=128
+  conv-relu-batchnorm-layer name=cnn6 $cnn_opts height-in=20 height-out=20  time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=128
```

Comment:
sorry, I didn't look at this before.  Can you try a version where the height-out of cnn5 and cnn6 is 10, not 20, and their num-filters-out is 256?  This will leave the compute time about the same (while increasing the parameters), and will allow those layers to see a wider range of frequency.  So reducing the height (and increasing the num-filters) actually increases the modeling power.

Reply:
OK will try

Reply:
tdnn7q_sp  cnn_tdnn1a_sp  cnn_tdnn1a_more_filters_sp
 WER on train_dev(tg)      12.08     12.13     11.97
 WER on train_dev(fg)      11.15     11.16     11.12
 WER on eval2000(tg)        14.1      14.1      13.9
 WER on eval2000(fg)        12.8      12.6      12.5
 WER on rt03(tg)            17.5      17.3      17.1
 WER on rt03(fg)            15.3      14.9      14.9
 Final train prob         -0.055    -0.057    -0.056
 Final valid prob         -0.072    -0.075    -0.075
 Final train prob (xent)        -0.875    -0.877    -0.871
 Final valid prob (xent)       -0.9064   -0.9134   -0.9110
 Num-parameters               18725244  14597020  15187100

Reply:
Great!  So that's the setup with more filters and height-out=10 on the last 2 layers, then?
In that case I think you should just change your 1a to be that configuration, and we could mege that.